src/libs/Violations/ViolationsUtils.ts

```diff
@@ -49,31 +50,67 @@ const ViolationsUtils = {
-            if (!hasTagOutOfPolicyViolation && updatedTransaction.tag && !isTagInPolicy) {
-                newTransactionViolations.push({name: 'tagOutOfPolicy', type: 'violation', userMessage: ''});
-            }
+            const selectedTags = updatedTransaction.tag?.split(CONST.COLON) ?? [];
+            const policyTagKeys = Object.keys(policyTagList);
 
-            // Remove 'tagOutOfPolicy' violation if tag is in policy
-            if (hasTagOutOfPolicyViolation && updatedTransaction.tag && isTagInPolicy) {
-                newTransactionViolations = reject(newTransactionViolations, {name: 'tagOutOfPolicy'});
+            if (policyTagKeys.length === 0) {
+                newTransactionViolations.push({
+                    name: CONST.VIOLATIONS.TAG_OUT_OF_POLICY,
+                    type: 'violation',
+                    userMessage: '',
+                });
             }
 
-            // Remove 'missingTag' violation if tag is valid according to policy
-            if (hasMissingTagViolation && isTagInPolicy) {
-                newTransactionViolations = reject(newTransactionViolations, {name: 'missingTag'});
-            }
+            policyTagKeys.forEach((key, index) => {
+                const hasTagOutOfPolicyViolation = transactionViolations.some((violation) => violation.name === CONST.VIOLATIONS.TAG_OUT_OF_POLICY && violation.data?.tagName === key);
+                const hasMissingTagViolation = transactionViolations.some((violation) => violation.name === CONST.VIOLATIONS.MISSING_TAG && violation.data?.tagName === key);
+                const selectedTag = selectedTags[index];
+                const isTagInPolicy = Boolean(policyTagList[key]?.tags[selectedTag]?.enabled);
 
-            // Add 'missingTag violation' if tag is required and not set
-            if (!hasMissingTagViolation && !updatedTransaction.tag && policyRequiresTags) {
-                newTransactionViolations.push({name: 'missingTag', type: 'violation', userMessage: ''});
-            }
+                // Add 'tagOutOfPolicy' violation if tag is not in policy
+                if (!hasTagOutOfPolicyViolation && selectedTag && !isTagInPolicy) {
+                    newTransactionViolations.push({
+                        name: CONST.VIOLATIONS.TAG_OUT_OF_POLICY,
+                        type: 'violation',
+                        userMessage: '',
+                        data: {
+                            tagName: key,
+                        },
+                    });
+                }
+
+                // Remove 'tagOutOfPolicy' violation if tag is in policy
+                if (hasTagOutOfPolicyViolation && selectedTag && isTagInPolicy) {
+                    newTransactionViolations = reject(newTransactionViolations, {
+                        name: CONST.VIOLATIONS.TAG_OUT_OF_POLICY,
+                        data: {
+                            tagName: key,
+                        },
```

Comment:
This broke the client side violation logic for tags -- I'm fixing it. Why did we add `{data { tagName }}` here, and why did we add `userMessage`?